### ISSUE TITLE
SEO - Craft 4 Support

### ISSUE THREAD

**Issue by BenParizek** [barrelstrength/sprout#190]
Sprout SEO is in the process of being updated to Craft 4. Please see the statement in our docs for additional details:
https://sprout.barrelstrengthdesign.com/docs/support/road-map.html

Follow this thread if you'd like to be notified when the upgrade is ready.

<img src="https://front.com/assets/img/favicons/favicon-32x32.png" height="16" width="16" alt="Front logo" /> [Front conversations](https://app.frontapp.com/open/top_55pql)

**Comment by anbod-io** [comment]
I have 5 sites which want to update to Craft 4, all of which are waiting on this plugin. 
Could we possibly get an update on progress?

**Comment by BenParizek** [comment]
The [Road Map](https://sprout.barrelstrengthdesign.com/docs/support/road-map.html) remains the best summary of where things stand. I last updated it on 10/25.

**Comment by anbod-io** [comment]
Oh, darn, that's disappointing. 
Sprout SEO was my favorite of the available SEO plugins and I wanted to keep using it.

Craft 4 compability has bin "in progress" since Craft 4 was released GA half a year ago and it doesn't seem like anything will change in the coming half year, or you would have mentioned it.

Well, thanks for a nice plug-in during my Craft 3 work, I enjoyed using it.

**Comment by BenParizek** [comment]
Sprout SEO will be updated with Craft 4 support and I understand the delays may put some projects in a position to consider alternatives in the interim. I'll update the Road Map with more specific details as I can commit to them.

**Comment by harryfk** [comment]
Heya, any updates on the progress, I saw the roadmap was last updated almost two months ago and it still mentions migration might be complete by early 2023. Is that somewhat realistic still?

My intention is not to put pressure on you, take all the time you need until it’s done, Ben! My unfortunate situation is just that a client project uses sprout-seo and they are quite eager to update to Craft 4, however, they also have hundreds of custom seo texts and images entered into sprout-seo fields, so migration is not trivial. Wondering if it’s worth the effort at this point or better to wait a few more months.

**Comment by BenParizek** [comment]
@harryfk I just got Sprout Redirects updated for Craft 4 this week, so we're moving in the right direction, but I expect the whole list of plugins may still take another few months before they are all ready.

I'm understanding if you need to make a decision to move forward in another way and if I can help with understanding the Sprout SEO data structure to prepare a migration, please just reach out to support.

My intention is to have all our commercial plugins updated to Craft 4 as soon as I can make that happen this year, while also making sure I reduce our code debt and get things in better shape so I can get back on track with more timely upgrades with Craft 5 and beyond.

**Comment by harryfk** [comment]
@BenParizek Excellent, thanks for the update and good to hear you’re actively working on it! Will just wait then and not worry about exporting/importing all that data. 🙏

**Comment by rebkuhn** [comment]
Hi. We have a lot of projects using the Sprout SEO Plugin and most of them should be updated to Craft 4 soon. Until now, we were able to postpone those updates because we always thought, the Craft 4 plugin update would be available soon. (We really love your plugins). But now, we need to make a "final" decision regarding this matter, as we cannot keep those projects on hold forever. In the Sprout Docs for Craft 3 you write, that you hope to get all modules of our plugins updated before the end of 2023. Is this schedule still realistic? If not, do you know what time range is realistic for you to finish the update? Kind Regards, Rebecca

**Comment by BenParizek** [comment]
@rebkuhn I was just working on the SEO plugin this past week and it's coming along and mostly working. I'm still working through a few things and I do think I'll be able to have it upgraded to Craft 4 this year (possibly in the next few weeks).

At this point, I expect the order of what remains will be:

- Sprout SEO (next C4 release)
- Sprout Forms (C4 beta release, still has a lot of details to sort out)
- Sprout Forms + Sprout Email (C4 betas -> C4 stable releases)
- Start upgrading to Craft 5!

While I'm hopeful that I'll have these releases out this year, and I think SEO is in a good place to start polishing at this point, I can't promise any specific dates. If that's not certain enough timing for you, I understand, and if I can help you understand anything about our plugin to make a migration easier, let me know.

**Comment by rebkuhn** [comment]
Dear @BenParizek,

I would like to inquire briefly whether you now have a definitive date for the release of The Craft 4 plugin updates.

Best regards, Rebecca

**Comment by BenParizek** [comment]
I've just released a beta of SEO for Craft 4.

Please read the upgrade notes as there are a few changes to the template tags:
https://sprout.barrelstrengthdesign.com/docs/craft-v4/update-guides/seo.html

**Comment by BenParizek** [comment]
We've removed the beta label for SEO v4 and work has begun on SEO compatibility for Craft 5 (#286).

**Comment by iamtompickering** [comment]
@BenParizek great to hear that the Craft 4 version has been released. I've just gone to upgrade 3.x site using Sprout SEO to 4.x though and hit an error with your migration for the Sprout redirects. Any reason why this would be happening?

**Comment by BenParizek** [comment]
@iamtompickering Can you open a new ticket and provide the error? The Sprout SEO plugin includes our meta, redirects, and sitemaps modules so it does make sense that those migrations would be running.
